File: fixed/src/main/java/com/hsf/fixed/MainActivity.java
package com.hsf.fixed;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import com.hsf.fixed.header_footer.HeaderFooterActivity;
import com.hsf.fixed.item_decoration.ItemDecorationActivity;
import com.hsf.fixed.item_decoration_two.ItemDecorationTwoActivity;
import com.hsf.fixed.one_view.OneViewActivity;
import com.hsf.fixed.two_same.TwoSameTopBarActivity;
import com.hsf.fixed.use_material.UseMaterialActivity;

public class MainActivity extends LogActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void onClickTwoTopBar(View view) {
        Intent intent = new Intent(MainActivity.this, TwoSameTopBarActivity.class);
        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        startActivity(intent);
    }

    public void onClickToHeaderFooter(View view) {
        Intent intent = new Intent(MainActivity.this, HeaderFooterActivity.class);
        intent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);
        startActivity(intent);
    }

    public void onClickOneTopBar(View view) {
        Intent intent = new Intent(MainActivity.this, OneViewActivity.class);
        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
        startActivity(intent);
    }

    public void onClickMaterialDesign(View view) {
        Intent intent = new Intent(MainActivity.this, UseMaterialActivity.class);
        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_SINGLE_TOP);
        startActivity(intent);
    }

    public void onClickItemDecoration(View view) {
        Intent intent = new Intent(MainActivity.this, ItemDecorationActivity.class);
        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK | Intent.FLAG_ACTIVITY_NEW_TASK);
        startActivity(intent);
    }

    public void onClickIemDecoration2(View view) {
        Intent intent = new Intent(MainActivity.this, ItemDecorationTwoActivity.class);
        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_NEW_TASK);
        startActivity(intent);
    }
}

File: coordinator/src/main/java/com/hsf/coordinator/one/MainActivity.kt
package com.hsf.coordinator.one

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.hsf.coordinator.ItemEntity
import com.hsf.coordinator.R
import com.hsf.coordinator.SubjectAdapter

class MainActivity : AppCompatActivity() {

    private val recyclerView: RecyclerView by lazy {
        findViewById(R.id.recyclerView)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        supportActionBar?.hide()

        val itemList = ArrayList<ItemEntity>()
        val strings = arrayOf("Android", "后端", "前端", "iOS", "人工智能", "产品", "工具资源", "阅读", "设计", "Android", "后端", "前端", "iOS", "人工智能", "产品", "工具资源", "阅读", "设计")
        strings.forEach {
            itemList.add(ItemEntity(it))
        }

        val linearLayoutManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)
        val subjectAdapter = SubjectAdapter(itemList)

        recyclerView.apply {
            layoutManager = linearLayoutManager
            adapter = subjectAdapter
        }
    }
}